dask/highlevelgraph.py

```diff
@@ -1053,6 +1054,112 @@ def __dask_distributed_unpack__(hlg: dict) -> dict:
+                    56.25L30.5 57.75C33.6756 57.75 36.25 55.1756 36.25 52L34.75 52ZM40.5 36.25C37.3244 36.25
+                    34.75 38.8243 34.75 42L36.25 42C36.25 39.6528 38.1528 37.75 40.5 37.75L40.5 36.25Z"
+                    fill="black"/>
+                <circle cx="28" cy="16" r="2.25" fill="#E5E5E5" stroke="black" stroke-width="1.5"/>
+                <circle cx="28" cy="57" r="2.25" fill="#E5E5E5" stroke="black" stroke-width="1.5"/>
+                <path d="M45.25 36.567C45.5833 36.7594 45.5833 37.2406 45.25 37.433L42.25 39.1651C41.9167
+                    39.3575 41.5 39.117 41.5 38.7321V35.2679C41.5 34.883 41.9167 34.6425 42.25 34.8349L45.25
+                    36.567Z" fill="#1D1D1D"/>
+            </svg>"""
+        html = f"""
+        <div style="">
+            <div>
+                <div style="
+                    width: 48px;
+                    height: 48px;
+                    border-radius: 5px;
+                    position: absolute;"> {highlevelgraph_icon} </div>
+                <div style="margin-left: 62px;">
+                    <h3 style="margin-bottom: 0px;">HighLevelGraph</h3>
+                    <p style="color: #5D5851; margin-bottom:
+        0px;">{highlevelgraph_info}</p>
+
+        <div style="">
+            <details open style="margin-left: 0px;">
+                <summary style="margin-bottom: 0px; margin-top: 5px;">
+                <h3 style="display: inline;">Layers</h3>
+            </summary>
+        """
+        for i, (key, layer) in enumerate(self.layers.items()):
+            key_shortname = key.rsplit("-", 1)[0]  # removes token
+            dependencies = self.dependencies[key]
+            info = layer_info_dict(layer, dependencies)
+            layer_info_table = html_from_dict(info)
+            if layer.is_materialized():
+                layer_icon = materialized_layer_icon
+            else:
+                layer_icon = unmaterialized_layer_icon
+
+            layer_html = f"""
+            <div style="">
+                <details style="margin-left: 0px;">
+                <summary style="margin-bottom: 20px; margin-top: 20px;">
+                    <div style="
+                        width: 48px;
+                        height: 48px;
+                        border-radius: 5px;
+                        position: absolute;"> {layer_icon} </div>
+                    <h3 style="display: inline; margin-left: 24px">
+                        Layer {i}: {key_shortname}</h3>
+                </summary>
+                <p style="color: #5D5851; margin-bottom: 0px;">{key}</p>
+                {layer_info_table}
+                </details>
+            </div>
+            """
+            html += layer_html
+        self.html = html
+        return html
+
+
+def layer_info_dict(layer, dependencies):
+    info = {
+        "layer_type": type(layer).__name__,
+        "is_materialized": layer.is_materialized(),
+        "dependencies": dependencies,
+    }
+    return info
```

Comment:
Not important, but my preference would be to keep logic like this inlined in order to avoid indirection (blog post here: https://matthewrocklin.com/blog/work/2019/06/23/avoid-indirection )

However, really I'd prefer that we move this down to Layer subclasses (even more indirection, I acknowledge).

Reply:
I agree, it should go with the Layer.

I don't agree that it should be inlined with the `_repr_html_` function logic, because imo that will make it harder for people making representations for layer subclasses. (Perhaps I misunderstand your point here though)

For someone writing a `_repr_html` function for a Layer subclass, it seems a lot easier to grab the default dictionary of interesting layer information with this function,, then add more dictionary keys/values specific to that particular Layer subclass, and then turn that into a nice HTML string. If you only have the HTML string, you might end up either duplicating a lot of the logic here for each layer subclass, OR faffing around removing HTML div tags, and trying to insert the new extra information into the right place in the HTML string.

Reply:
I agree that dupliation would ensure (we would have to address this with methods, functions, etc.. 

However it could open up larger variations than just different keys/values in the table.  For example we might include images like the chunk structure, or different coloring based on layer type.